crates/signer/src/signer.rs

```diff
@@ -113,4 +146,123 @@ mod test {
 
         assert_eq!(signature, expected_signature);
     }
+
+    // This test will fail if docker is not running
+    #[tokio::test]
+    async fn sign_transaction_with_aws_signer() {
+        // Start the container running Localstack
+        let _container = start_localstack_container().await;
+
+        let localstack_endpoint = format!("http://localhost:{}", LOCALSTACK_PORT);
+        let config = get_aws_config(
+            "localstack".into(),
+            "localstack".into(),
+            Region::from_static(&AWS_SOUTH_AMERICA_REGION),
+            localstack_endpoint,
+        )
+        .await;
+
+        // Create an AWS KMS Client
+        let client = aws_sdk_kms::Client::new(&config);
+
+        // Create a key
+        let key_metadata = create_kms_key(&client).await;
+
+        // Create a signer for the given key
+        let key_id = key_metadata.key_id();
+        let chain_id = Some(1);
+        let signer = Config::aws_signer(key_id.into(), chain_id, client.clone())
+            .await
+            .unwrap();
+
+        let mut tx = TxLegacy {
+            to: address!("d8dA6BF26964aF9D7eEd9e03E53415D37aA96045").into(),
+            value: U256::from(1_000_000_000),
+            gas_limit: 2_000_000,
+            nonce: 0,
+            gas_price: 21_000_000_000,
+            input: bytes!(),
+            chain_id: Some(1),
+        };
+
+        // Sign the transaction
+        let signature = signer.sign_transaction(&mut tx).await.unwrap();
+
+        // Recover the address
+        let mut encoded_tx = Vec::new();
+        tx.encode_for_signing(&mut encoded_tx);
+        let prehash = keccak256(encoded_tx);
+        let recovered_address = signature.recover_address_from_prehash(&prehash).unwrap();
+
+        // Check that the recovered addresses are the same
+        assert_eq!(signer.address(), recovered_address);
+    }
+
+    // This test will fail if anvil is not installed (Foundry contains anvil)
```

Comment:
could you add this and the docker prereqs to run tests to the docs?

Reply:
do you mean the rust docs comments? or should we add a `README.md` with the signer docs?

Reply:
I meant to add this to the parent/root readme, since these tests run with the rest of the tests and these pre-reqs are required for the tests to pass